incalmo/c2server/state_store.py

```diff
@@ -1,55 +1,40 @@
 import json
 import os
+import sqlite3
 from typing import Any, Optional
 
-from redis import Redis
-
-
 class StateStore:
-    _redis_client: Optional[Redis] = None
-    _memory_cache: dict[str, Any] = {"environment:hosts": []}
-
-    @classmethod
-    def _get_redis_client(cls) -> Optional[Redis]:
-        if cls._redis_client is not None:
-            return cls._redis_client
-
-        # Prefer explicit state store URL, then Celery backend/broker, then default
-        redis_url = (
-            os.environ.get("STATE_REDIS_URL")
-            or os.environ.get("result_backend")
-            or os.environ.get("broker_url")
-            or "redis://localhost:6379/0"
-        )
-
-        client = Redis.from_url(redis_url, decode_responses=True)
-        # Validate connection
-        client.ping()
-        cls._redis_client = client
-        return cls._redis_client
+    TABLE_NAME = "environment"
+    DB_PATH = "state_store.db"
 
     @classmethod
     def set_hosts(cls, hosts: list[dict]) -> None:
-        client = cls._get_redis_client()
-        if client is not None:
-            client.set("environment:hosts", json.dumps(hosts))
-            return
-        cls._memory_cache["environment:hosts"] = hosts
+        cls._db_connection = sqlite3.connect(cls.DB_PATH)
+        cursor = cls._db_connection.cursor()
+        cursor.execute(
+            f"""
+            CREATE TABLE IF NOT EXISTS {cls.TABLE_NAME} (
```

Comment:
This is kinda jank lol we will need a better way of managing this state in the future, but OK for this PR